src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -0,0 +1,212 @@
+        self,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        self.__closed = False
+        self.__futures: Deque[ReplayRecordingMessageFuture] = deque()
+        self.__threadpool = concurrent.futures.ThreadPoolExecutor()
+        self.__commit = commit
+
+    def poll(self) -> None:
+        self._commit_and_prune_futures()
+
+    def _process_chunk(
+        self, message_dict: RecordingSegmentChunkMessage, message: Message[KafkaPayload]
+    ) -> None:
+        # TODO: implement threaded chunk sets, and wait for an individual segment's
+        # futures to finish before trying to read from redis in the final kafka message
+        # https://github.com/getsentry/replay-backend/pull/38/files
+        recording_segment_uuid = message_dict["id"]
+        project_id = message_dict["project_id"]
+        chunk_index = message_dict["chunk_index"]
+        cache_key = replay_recording_segment_cache_id(project_id, recording_segment_uuid)
+
+        attachment_cache.set_chunk(
+            key=cache_key,
+            id=recording_segment_uuid,
+            chunk_index=chunk_index,
+            chunk_data=message_dict["payload"],
+            timeout=CACHE_TIMEOUT,
+        )
+
+    def _process_headers(
+        self, recording_segment_with_headers: bytes
+    ) -> tuple[RecordingSegmentHeaders, bytes]:
+        # split the recording payload by a newline into the headers and the recording
+        try:
+            recording_headers, recording_segment = recording_segment_with_headers.split(b"\n", 1)
+        except ValueError:
+            raise MissingRecordingSegmentHeaders
+        return json.loads(recording_headers), recording_segment
+
+    def _store(
+        self,
+        message_dict: RecordingSegmentMessage,
+        cached_replay_recording_segment: CachedAttachment,
+    ) -> None:
+        try:
+            headers, recording_segment = self._process_headers(cached_replay_recording_segment.data)
+        except MissingRecordingSegmentHeaders:
+            logger.warning(f"missing header on {message_dict['replay_id']}")
+            return
+
+        # create a File for our recording segment.
+        recording_segment_file_name = f"rr:{message_dict['replay_id']}:{headers['sequence_id']}"
+        file = File.objects.create(
+            name=recording_segment_file_name,
+            type="replay.recording",
+        )
+        file.putfile(
+            BytesIO(recording_segment),
+            blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
+        )
+        # associate this file with an indexable replay_id via ReplayRecordingSegment
+        ReplayRecordingSegment.objects.create(
+            replay_id=message_dict["replay_id"],
+            project_id=message_dict["project_id"],
+            sequence_id=headers["sequence_id"],
+            file_id=file.id,
+        )
+        # delete the recording segment from cache after we've stored it
+        cached_replay_recording_segment.delete()
```

Comment:
We should suppress exceptions here since this operation is ostensibly completed.

Reply:
we don't suppress exceptions elsewhere where we call delete -- can monitor and see if exceptions happen at all here.